bin/tag.ml

```diff
@@ -10,22 +10,54 @@ open Dune_release
 let vcs_tag tag ~dry_run ~commit_ish ~force ~sign ~delete ~msg ~yes =
   let msg = match msg with None -> strf "Distribution %s" tag | Some m -> m in
   Vcs.get () >>= fun repo ->
+  let tag_exists = Vcs.tag_exists ~dry_run:false repo tag in
   match delete with
   | true ->
-      Prompt.confirm_or_abort ~yes ~question:(fun l ->
-          l "Delete tag %a?" Text.Pp.version tag)
-      >>= fun () ->
-      Vcs.delete_tag ~dry_run repo tag >>| fun () ->
-      App_log.success (fun m -> m "Deleted tag %a" Text.Pp.version tag)
+      if tag_exists then
```

Comment:
Following my suggestion regarding `tag_points_to`, I think this whole function could be simplified a bit, maybe as a single pattern matching over `delete`, `commit_ish` and the optional commit the tag points to.

It is a bit convoluted at the moment and makes it hard to follow. What do you think?

Reply:
Yes, that actually makes it far nicer! :) Thanks.